chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -739,166 +740,221 @@ async def make_nft1_offer(
-                )
-                all_transactions = txs
-
-            txn_bundles: List[SpendBundle] = [tx.spend_bundle for tx in all_transactions if tx.spend_bundle is not None]
-            txn_spend_bundle = SpendBundle.aggregate(txn_bundles)
-            # Create a spend bundle for the royalty payout from OFFER MOD
-            # make the royalty payment solution
-            # ((nft_launcher_id . ((ROYALTY_ADDRESS, royalty_amount, (ROYALTY_ADDRESS)))))
-            # we are basically just recreating the royalty announcement above.
-            inner_royalty_sol = Program.to([[requested_asset_id, [royalty_address, royalty_amount, [royalty_address]]]])
-            if offered_asset_id is None:  # xch offer
-                offer_puzzle: Program = OFFER_MOD
-                royalty_sol = inner_royalty_sol
-            else:  # CAT
-                offer_puzzle = construct_puzzle(driver_dict[offered_asset_id], OFFER_MOD)
-            royalty_ph = offer_puzzle.get_tree_hash()
-            for txn in txn_bundles:
-                for coin in txn.additions():
-                    if coin.amount == royalty_amount and coin.puzzle_hash == royalty_ph:
-                        royalty_coin = coin
-                        parent_spend = txn.coin_spends[0]
-            assert royalty_coin
-            if offered_asset_id is not None:  # if CAT
-                #  adapt royalty_sol to work with cat puzzle
-                royalty_coin_hex = (
-                    "0x"
-                    + royalty_coin.parent_coin_info.hex()
-                    + royalty_coin.puzzle_hash.hex()
-                    + bytes(uint64(royalty_coin.amount)).hex()
-                )
-                parent_spend_hex: str = "0x" + bytes(parent_spend).hex()
-                solver = Solver(
-                    {
-                        "coin": royalty_coin_hex,
-                        "parent_spend": parent_spend_hex,
-                        "siblings": "(" + royalty_coin_hex + ")",
-                        "sibling_spends": "(" + parent_spend_hex + ")",
-                        "sibling_puzzles": "()",
-                        "sibling_solutions": "()",
-                    }
-                )
-                royalty_sol = solve_puzzle(driver_dict[offered_asset_id], solver, OFFER_MOD, inner_royalty_sol)
 
-            royalty_spend = SpendBundle([CoinSpend(royalty_coin, offer_puzzle, royalty_sol)], G2Element())
-            total_spend_bundle = SpendBundle.aggregate([txn_spend_bundle, royalty_spend])
-            offer = Offer(notarized_payments, total_spend_bundle, driver_dict)
-            return offer
-        else:
-            raise ValueError("No NFT in offer!")
+        # Create all of the transactions
+        all_transactions: List[TransactionRecord] = []
+        additional_bundles: List[SpendBundle] = []
+        # standard pays the fee if possible
+        fee_left_to_pay: uint64 = uint64(0) if None in offer_dict and offer_dict[None] < 0 else fee
```

Comment:
should this be checking if None in offer_dict.keys()?

Reply:
This is the default behavior on `in` for dictionaries